### ISSUE TITLE
Crash on first launch if blocked by firewall

### ISSUE THREAD

**Issue by rhazor** [Openarl/PathOfBuilding#50]
Debug:
- Have your firewall set to block all unknown/new applications from outbound TCP connections.
- Install the .exe installer of PathOfBuilding and launch it. It will give some error log on green window, not mentioning the connections issues though.
- Close it and add rule to firewall to allow the application connecting to the internet.
- Launch the application again.
- Nothing happens, no action, keep clicking the app shortcut on desktop it just won't do anything, no log, nothing in Task Manager.
- Rerun installer.
- Fixed.

Just thought people should know.

**Comment by Openarl** [comment]
The application does need internet access when it's run the first time, as it updates itself to the latest version after installation (this is why I don't bother releasing new installer versions). It should fail more gracefully when the connection is blocked though. I'll have to investigate it more when I get the time.

**Comment by Openarl** [comment]
The new installer versions that will be released soon will no longer force an update when the program is run for the first time, which will avoid issues like this. The program may still encounter problems updating itself if there's issues with, or restrictions on, its internet access; but you'll at least be able to run the version that's installed.

**Comment by Openarl** [comment]
Well, "soon" took a bit longer than I'd hoped, but the latest installer versions will have solved this. The standalone version no longer forces an update check when run for the first time; the normal installer still does but it won't brick itself if that update check fails.
